src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function roundedNum(num: number): number {
+  return Math.floor(num * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Enter the correct values for sides');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Can not build the triangle!');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt((semiPerimeter - this.a) * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c) * semiPerimeter);
 
+    return roundedNum(area);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
 
+  constructor(public color: Color, public radius: number) {
```

Comment:
```suggestion
  constructor(
    public color: Color, 
    public radius: number
  ) {
```
Maybe in this way, it looks more readable) Do U agree?

Reply:
agree